lttoolbox/fst_processor.cc

```diff
@@ -2998,101 +2619,98 @@ FSTProcessor::biltrans(wstring const &input_word, bool with_delim)
 
     if(with_delim)
     {
-      result_with_queue += L'$';
+      result_with_queue += '$';
     }
     return result_with_queue;
   }
   else
   {
     if(with_delim)
     {
-      result += L'$';
+      result += '$';
     }
     return result;
   }
 }
 
 void
-FSTProcessor::bilingual_wrapper_null_flush(FILE *input, FILE *output, GenerationMode mode)
+FSTProcessor::bilingual_wrapper_null_flush(InputFile& input, UFILE *output, GenerationMode mode)
 {
   setNullFlush(false);
   nullFlushGeneration = true;
 
-  while(!feof(input))
+  while(!input.eof())
   {
     bilingual(input, output, mode);
-    fputwc_unlocked(L'\0', output);
-    int code = fflush(output);
-    if(code != 0)
-    {
-        wcerr << L"Could not flush output " << errno << endl;
-    }
+    u_fputc('\0', output);
+    u_fflush(output);
   }
 }
 
-wstring
-FSTProcessor::compose(wstring const &lexforms, wstring const &queue) const
+UString
+FSTProcessor::compose(UString const &lexforms, UString const &queue) const
 {
-  wstring result = L"";
+  UString result;
+  result += '/';
 
   for(unsigned int i = 1; i< lexforms.size(); i++)
   {
-    if(lexforms[i] == L'\\')
+    if(lexforms[i] == '\\')
     {
-      result += L'\\';
+      result += '\\';
       i++;
     }
-    else if(lexforms[i] == L'/')
+    else if(lexforms[i] == '/')
     {
       result.append(queue);
     }
     result += lexforms[i];
   }
 
-  return L"/" + result + queue;
+  return result + queue;
```

Comment:
Better to `result += queue; return result;` to take advantage of https://en.cppreference.com/w/cpp/language/copy_elision
Also, can one make an educated guess at a max size of result to `.reserve()` it up front?